### DISCUSSION TITLE
[RFC] ColorPicker implement panelRender API

### DISCUSSION THREAD

**Discussion by RedJue** [ant-design/ant-design#42833] [RFCs - archive]
## Summary

Implement `panelRender` API for customizing the panel layout.

## Motivation

As  #42749 mentioned. 

Some users want to be able to customize the layout of the color picker panel, so that they can easily add  parts from the panel, although the color picker will provide `layout` API, but it does not meet the needs of user-defined panels.

Therefore, we need to implement an API that can facilitate user-defined panels.

## API

| Property   | Description | Type | 
| ---------- | ----------- | ------| 
| panelRender  |    custom panel render   |  (panel: ReactNode, extra: { components: { Picker: Component, Presets: Component } }) => ReactNode 

## Basic Example

```tsx
const Demo = () => (
  <ColorPicker
    panelRender={(panel, { components: { Picker, Presets }}) => {
      return (
        <div className="custom-panel">
          <Picker />
          <Presets />
        </div>
      );
    }}
  />
);
```
## How we achieved it

- Implement `panelRender` API at `antd`.
- Add demo and test case.

**Comment by kiner-tang** [comment]
是不是除了 panel 参数之外，还可以增加一个参数传入 内部的一些状态啥的，方便在渲染面板的时候用于判断。例如：
```tsx
const Demo = () => (
  <ColorPicker
    panelRender={(panel, info: {color?: Color}) => {
      return (
        <div className="custom-panel">
          {info.color?.toHexString()}
          <div className="base-panel"> {panel} </div>
        </div>
      );
    }}
  />
);
```

**Reply by RedJue** [reply]
好像都可以从外面拿？不知道需不需要内置。

**Reply by kiner-tang** [reply]
如果没有一些可能影响渲染的内部状态的话，感觉也可以先不加，以后如果有了再加也可以

**Reply by RedJue** [reply]
嗯嗯

**Comment by afc163** [comment]
```
    panelRender={({ panel }) => {
```

先改成这样把，方便后面加东西。

**Reply by RedJue** [reply]
ok
